app/models/office.rb

```diff
@@ -8,4 +8,71 @@ class Office < ApplicationRecord
   validates :region, numericality: { only_integer: true }
 
   accepts_nested_attributes_for :address, update_only: true
+
+  scope :with_claimed_shifts, -> { joins(needs: :shifts).merge(Shift.claimed) }
+  scope :with_claimed_needs, -> { joins(:needs).merge(Need.has_claimed_shifts) }
+
+  def self.claimed_shifts_by_office
```

Comment:
This looks **nice**! I do wonder if we should consider moving all these class methods into a module of some sort. If they were only used for admin reporting, perhaps something like `OfficeAdminReportMethods`. 

I don't think this needs to be done now, but something to consider 🤔